src/mesh/curved_mesh.jl

```diff
@@ -273,14 +193,62 @@ function trilinear_mapping(x, y, z, mesh)
 end
 
 
+# Use linear mapping in 1D
+transfinite_mapping(faces::NTuple{2, Any}) = x -> linear_mapping(x, faces)
+
 # In 2D
 # Transfinite mapping from the reference element to the domain described by the faces
-function transfinite_mapping(x, y, mesh)
-  @unpack faces = mesh
+function transfinite_mapping(faces::NTuple{4, Any})
+  mapping(x, y) = linear_interpolate(x, faces[1](y), faces[2](y)) + 
+                  linear_interpolate(y, faces[3](x), faces[4](x)) - 
+                  bilinear_mapping(x, y, faces)
+end
+
+# TODO: Use transfinite mapping instead of trilinear mapping to allow curved faces
+transfinite_mapping(faces::NTuple{6, Any}) = (x, y, z) -> trilinear_mapping(x, y, z, faces)
+
+
+function validate_faces(faces::NTuple{2, Any}) end
 
-  return linear_interpolate(x, faces[1](y), faces[2](y)) + 
-         linear_interpolate(y, faces[3](x), faces[4](x)) - 
-         bilinear_mapping(x, y, mesh)
+function validate_faces(faces::NTuple{4, Any})
+  @assert faces[1](-1) ≈ faces[3](-1) "faces[1](-1) needs to match faces[3](-1) (bottom left corner)"
+  @assert faces[2](-1) ≈ faces[3]( 1) "faces[2](-1) needs to match faces[3](1) (bottom right corner)"
+  @assert faces[1]( 1) ≈ faces[4](-1) "faces[1](1) needs to match faces[4](-1) (top left corner)"
+  @assert faces[2]( 1) ≈ faces[4]( 1) "faces[2](1) needs to match faces[4](1) (top right corner)"
+end
+
+function validate_faces(faces::NTuple{6, Any})
+  @assert faces[1](-1, -1) ≈ 
+          faces[3](-1, -1) ≈ 
+          faces[5](-1, -1) "faces[1](-1, -1), faces[3](-1, -1) and faces[5](-1, -1) need to match at (-1, -1, -1) corner"
```

Comment:
Can you please add parens (as suggested or in a different way)? This implicit line continuation has already been a source of subtle bugs in Trixi and I'd rather not rely on all users understanding this...
```suggestion
  @assert (faces[1](-1, -1) ≈ 
           faces[3](-1, -1) ≈ 
           faces[5](-1, -1)) "faces[1](-1, -1), faces[3](-1, -1) and faces[5](-1, -1) need to match at (-1, -1, -1) corner"
```